Updating data sources docs # 7680 (#7800)
opensearch-project/documentation-website: _dashboards/management/connect-prometheus.md

---
layout: default
title: Connecting Prometheus to OpenSearch
parent: Data sources
nav_order: 20
---

> # Connecting Prometheus to OpenSearch

Reviewer: We need to add visuals to this documentation. It is end user doc and dashboards related. See http://localhost:4000/docs/latest/dashboards/management/S3-data-source/ for an example.

> 

Reviewer: @AntonEliatra Please let me know if you cut this content from my previously published content. It appears to be a cut and paste from https://opensearch.org/docs/latest/dashboards/management/multi-data-sources/#step-2-create-a-new-data-source-connection. Let me know which text actually was added or edited.

Author: Hi @vagimeli, I'm not sure which content you are referring to, My commits are available in this PR, the very first one is https://github.com/opensearch-project/documentation-website/pull/7800/commits/9a13153fcd0c66154fd7a12e336f55410f05bb8b

A lot of the parts were moved around

Reviewer: @AntonEliatra I'll take over this PR as the tech writer to and work with the feature's developer to address information gaps. Thanks for writing the initial content.

> This documentation focuses on using the OpenSeach Dashboards interface to connect to Prometheus. 

Reviewer 2: "This page shows you how to use the..."?

## Prerequisites and permissions

> Make sure you followed and configured [Prerequisites]({{site.url}}{{site.baseurl}}/dashboards/management/data-sources/#prerequisites) and [Permissions]({{site.url}}{{site.baseurl}}/dashboards/management/data-sources/#permissions).

Reviewer 2: I believe there may be boilerplate for this elsewhere that we should be consistent with. It should read something like "Make sure that you have configured all necessary prerequisites and permissions for/in order to..."

> Ensure you have configured the necessary [Prerequisites]({{site.url}}{{site.baseurl}}/dashboards/management/data-sources/#prerequisites) and [Permissions]({{site.url}}{{site.baseurl}}/dashboards/management/data-sources/#permissions).

Reviewer: ```suggestion
Before connecting a data source, ensure you have met the [Prerequisites]({{site.url}}{{site.baseurl}}/dashboards/management/data-sources/#prerequisites) and have the necessary [Permissions]({{site.url}}{{site.baseurl}}/dashboards/management/data-sources/#permissions).
```

## Creating a Prometheus data source connection

A data source connection specifies the parameters needed to connect to a data source. These parameters form a connection string for the data source. Using Dashboards, you can add new **Amazon S3** or **Prometheus** data source connections or manage existing ones.

The following steps guide you through the basics of creating a **Prometheus** data source connection:

1. From the OpenSearch Dashboards main menu, select **Management** > **Data sources** > **New data source** > **Prometheus**. 

> 2. Create the data source connection by entering the appropriate information into the **Configure Prometheus data source** fields. 

Reviewer 2: ```suggestion
2. Create the data source connection by entering the appropriate information in the **Configure Prometheus data source** fields. 
```

Reviewer: Revised sentence to eliminate wordiness

>    - Under **Data source details**, enter a title and optionally description for this data source.

Reviewer: ```suggestion
   - Under **Data source details**, enter a title and, optionally, a description for this data source.
```

>    - Under **Data source details**, enter a title and, optionally, a description for this data source.

Reviewer: ```suggestion
   - Under **Data source details**, enter a title for this data source. Entering a description is optional.
```

>    - Under **Data source details**, enter a title for this data source. Entering a description is optional.

Reviewer 2: ```suggestion
   - Under **Data source details**, enter a title for the data source. Entering a description is optional.
```

>    - Under **Prometheus data location**, enter the Prometheus URI

Reviewer: ```suggestion
   - Under **Prometheus data location**, enter the Prometheus URI.
```

       - **Basic authentication**: Enter a username and password.

>        - **AWS Signature Version 4**: Specify the specify the **Region**, select the OpenSearch service from the **Service Name** list (**Amazon OpenSearch Service** or **Amazon OpenSearch Serverless**), and enter the **Access Key** and **Secret Key**.

Reviewer: ```suggestion
       - **AWS Signature Version 4**: Specify the **Region**, select the OpenSearch service from the **Service Name** list (**Amazon OpenSearch Service** or **Amazon OpenSearch Serverless**), and enter the **Access Key** and **Secret Key**.
```

       - **AWS Signature Version 4**: An AWS Signature Version 4 authenticating request is used to connect to the data source. AWS Signature Version 4 requires an access key and a secret key. 
         - For AWS Signature Version 4 authentication, first specify the **Region**. Next, select the OpenSearch service in the **Service Name** list. The options are **Amazon OpenSearch Service** and **Amazon OpenSearch Serverless**. Lastly, enter the **Access Key** and **Secret Key** for authorization.

>    - Under **Query permissions**, choose which role is needed to search and index data from this data source. If **Restricted** is selected, additional field will become available which is used to configure the required role.

Reviewer 2: ```suggestion
   - Under **Query permissions**, choose which role is needed in order to search and index data from this data source. If **Restricted** is selected, an additional field will become available that is used to configure the required role.
```

Reviewer: Revised sentence to eliminate wordiness

> 3. When you have entered all the details, select **Review Configuration** > **Connect to Prometheus** to save your settings. The connection is created. The active window returns to the **Data sources** main page, and the new connection appears in the list of data sources.

Reviewer 2: ```suggestion
3. When you have entered all the information, select **Review Configuration** > **Connect to Prometheus** to save your settings. The connection is created. The active window returns to the **Data sources** main page, and the new connection appears in the list of data sources.
```

Reviewer: Revised sentence to eliminate wordiness

### Modifying a data source connection

> To make changes to a data source connection, select a connection in the list on the **Data sources** main page. The **Connection Details** window opens.

Reviewer 2: ```suggestion
To make changes to a data source connection, select a connection from the list on the **Data sources** main page. The **Connection Details** window opens.
```

Reviewer: Revised sentence to eliminate wordiness

To update the **Authentication Method**, choose a different method from the dropdown list, enter the required credentials (if applicable), and then select **Save changes** to apply the changes.

> If the selected authentication methos is **Username & Password**, you can update the password by choosing **Update stored password** next to the **Password** field. In the pop-up window, enter the new password and then confirm it, then select **Update stored password** to save the changes. Select **Test connection** to confirm that the connection is valid.

Reviewer: ```suggestion
If the selected authentication method is **Username & Password**, you can update the password by choosing **Update stored password** next to the **Password** field. In the pop-up window, enter the new password and then confirm it, then select **Update stored password** to save the changes. Select **Test connection** to confirm that the connection is valid.
```

> If the selected authentication method is **Username & Password**, you can update the password by choosing **Update stored password** next to the **Password** field. In the pop-up window, enter the new password and then confirm it, then select **Update stored password** to save the changes. Select **Test connection** to confirm that the connection is valid.

Reviewer: ```suggestion
If the selected authentication method is **Username & Password**, you can update the password by choosing **Update stored password** next to the **Password** field. In the pop-up window, enter the new password and confirm it, then select **Update stored password** to save the changes. Select **Test connection** to confirm that the connection is valid.
```

For **AWS Signature Version 4** authentication, you can update the credentials by selecting **Update stored AWS credential**. In the pop-up window, enter the new access key and the new secret key. Select **Update stored AWS credential** to save the changes. Select **Test connection** to confirm that the connection is valid.

To delete the data source connection, select the {::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/dashboards/trash-can-icon.png" class="inline-icon" alt="delete icon"/>{:/} icon.

## Creating an index pattern

> After creating a data source connection, the next step is to create an index pattern for that data source. For more information and a tutorial on index patterns. refer to [Index patterns]({{site.url}}{{site.baseurl}}/dashboards/management/index-patterns/). 

Reviewer: ```suggestion
After creating a data source connection, the next step is to create an index pattern for that data source. For more information and a tutorial on index patterns, refer to [Index patterns]({{site.url}}{{site.baseurl}}/dashboards/management/index-patterns/). 
```

---
layout: default
title: Connecting Prometheus to OpenSearch
parent: Data sources
nav_order: 20
---

# Connecting Prometheus to OpenSearch
Introduced 2.16
{: .label .label-purple }

This documentation covers the key steps to connect Prometheus to OpenSearch using the OpenSearch Dashboards interface, including setting up the data source connection, modifying the connection details, and creating an index pattern for the Prometheus data. 

## Prerequisites and permissions

Before connecting a data source, ensure you have met the [Prerequisites]({{site.url}}{{site.baseurl}}/dashboards/management/data-sources/#prerequisites) and have the necessary [Permissions]({{site.url}}{{site.baseurl}}/dashboards/management/data-sources/#permissions).

## Create a Prometheus data source connection

> A data source connection specifies the parameters needed to connect to a data source. These parameters form a connection string for the data source. Using OpenSearch Dashboards, you can add new **Amazon S3** or **Prometheus** data source connections or manage existing ones.

Reviewer: ```suggestion
A data source connection specifies the parameters needed to connect to a data source. These parameters form a connection string for the data source. Using OpenSearch Dashboards, you can add new **Prometheus** data source connections or manage existing ones.
```

> Follow these steps to connect **Prometheus** to OpenSearch:

Reviewer: ```suggestion
Follow these steps to connect your data source:
```
